trin-history/src/validation.rs

```diff
@@ -162,7 +151,7 @@ mod tests {
         let hwp = HeaderWithProof::from_ssz_bytes(&hwp_ssz).expect("error decoding header");
```

Comment:
Seems worthwhile at least adding a simple validate/invalidate test for `BlockHeaderWithNumber` here